### PR TITLE
compares binary to text before returning the in memory representation.

### PR DESCRIPTION
**Pull request by wesboyt**
Makes sure that the binary serialization is equal to the text serialization on read.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonEventStream.ts [hunk 1]
```diff
@@ -301,6 +299,22 @@ export class IonEventStream {
             currentEvent['isNull'],
             currentEvent['value_text']
         );
+
+        let binaryEvent = this.eventFactory.makeEvent(
```
Comment (tgregg) [inline src/IonEventStream.ts]: This can go inside the `if(eventType === IonEventType.SCALAR){` block below since it's only needed in that case. To simplify it, you could also just do a data model equivalence comparison on `value_text` and `value_binary` if it's a scalar event. That way you don't need to create a dummy event for binary.

**Comment by wesboyt** [comment]
> The change looks good, but could you comment on the impact of this PR? For example: does this allow us to remove tests from our skip lists? Does it make all of the existing event stream tests more thorough?

This does not allow us to remove tests from our skip lists afaik, it does increase the possible edge cases we can catch when reading eventstreams from other serializers than our own.
